api/db.py

```diff
@@ -49,21 +58,101 @@ def get_single_data(table, field, value):
     with get_db_cursor() as cur:
         cur.execute(sql)
         data = cur.fetchone()
+        if data is None:
+            return TypeError('No such data exists!')
     return data
 
 
-# get all known info about current minister or kns member
-def get_fully_today_member(query, value):
+# get all known info about current minister or knesset member
+def get_fully_today_member(query, value=None):
     try:
         with get_db_cursor() as cur:
             cur.execute(query, value)
             data = cur.fetchone()
-    except Exception:
-        return ValueError('No such member exist!')
+            if data is None:
+                raise ValueError('No such member exist!')
+    except Exception as e:
+        return e
     return data
 
+# Get info about knesset members
+# is_current = true --> current Knesset members
+# is_current = false --> inactive Knesset members
+# knesset_num --> Knesset period of the Knesset member
+def get_members(query,is_current:bool,knesset_num:int):
+    try:
+        with get_db_cursor() as cur:
+            if is_current:
+                if(knesset_num==None):
+                    cur.execute('select max("KnessetNum") from knesset_kns_knessetdates')
+                    knesset_number = cur.fetchone()
+                    is_current_params=(knesset_number['max'],
+                                 knesset_number['max'],
+                                 knesset_number['max'],
+                                 knesset_number['max'],
+                                 is_current)
+                    cur.execute(query,is_current_params)
+                    data = cur.fetchall()
+                else:
+                    is_current_params=(knesset_num,
+                                 knesset_num,
+                                 knesset_num,
+                                 knesset_num,
+                                 is_current)
+                    cur.execute(query,is_current_params)
+                    data = cur.fetchall()
+               
+            else:
+                is_current_params=(knesset_num,
+                                 knesset_num,
+                                 knesset_num,
+                                 knesset_num,
+                                 is_current)
+                cur.execute(query,is_current_params)
+                data = cur.fetchall()
+    except Exception as e:
+        return e
+    return data
+
+# Get info about some Knesset member
+def get_members_info(query,mk_individual_id:int): 
+    try:
+        with get_db_cursor() as cur:
+            cur.execute(query)
+            data = cur.fetchall()
+            # Two options:
+            # 1. No such 'mk_individual_id' in database
+            # 2. 'mk_individual_id' exist but no data for him
+            if (not(is_mk_individual_exist(mk_individual_id)) 
+                or len(data) == 0 ):
```

Comment:
So can I put `logger.critical("critical error",e)`  inside `except Exception as e:` in 
`is_mk_individual_exist` ? And then it will be ok as it will return `None`.
Also, does it better to divide the if statement to two seperate statements?:
           # No such 'mk_individual_id' in database
            if not (is_mk_individual_exist(mk_individual_id)):
                return TypeError('No such data exists! Please enter another ID')
            # 'mk_individual_id' exist but no data for it
            elif len(data) == 0:
                return ValueError('This member has no data currently!')